File: Model/src/main/java/com/example/network/dto/NetworkSnapshot.java
package com.example.network.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class NetworkSnapshot {
    private LocalDateTime snapshotTime;
    private GpsData location;
    private CellularNetwork cellularNetwork;
    private List<WifiNetwork> wifiNetworks;
    private List<BluetoothDeviceInfo> bluetoothDevices;
}

File: client/app/src/main/java/com/example/gpslessclient/model/NetworkSnapshot.kt
package com.example.gpslessclient.model


data class NetworkSnapshot(
    val snapshotTime: String?,
    val location: GpsData?,
    val wifiNetworks: List<WifiNetwork> = emptyList(),
    val cellularNetwork: CellularNetwork?,
    val bluetoothDevices: List<BluetoothDeviceInfo> = emptyList()
)